lib/spack/spack/mirror.py

```diff
@@ -271,11 +331,57 @@ def get_url(self, direction: str) -> str:
 
         return _url_or_path_to_url(url)
 
-    def get_access_token(self, direction: str) -> Optional[str]:
-        return self._get_value("access_token", direction)
+    def get_credentials(self, direction: str) -> Dict[str, Any]:
+        """Get the mirror credentials from the mirror config
+
+        Args:
+            direction: fetch or push mirror config
+
+        Returns:
+            Dictionary from credential type string to value
+
+            Credential Type Map:
+                access_token -> str
+                access_pair  -> tuple(str,str)
+                profile      -> str
+        """
+        creddict: Dict[str, Any] = {}
+        access_token = self.get_access_token(direction)
+        if access_token:
+            creddict["access_token"] = access_token
+
+        access_pair = self.get_access_pair(direction)
+        if access_pair:
+            creddict.update({"access_pair": access_pair})
+
+        profile = self.get_profile(direction)
+        if profile:
+            creddict["profile"] = profile
+
+        return creddict
+
+    @staticmethod
+    def _extract_credential_value(tok):
+        return os.environ.get(tok["variable"]) if isinstance(tok, dict) else tok
 
-    def get_access_pair(self, direction: str) -> Optional[List]:
-        return self._get_value("access_pair", direction)
+    def get_access_token(self, direction: str) -> Optional[str]:
+        tok = self._get_value("access_token_variable", direction)
+        if tok:
+            return os.environ.get(tok)
+        else:
+            return self._get_value("access_token", direction)
+        return None
+
+    def get_access_pair(self, direction: str) -> Optional[Tuple[str, str]]:
+        pair = self._get_value("access_pair", direction)
+        if type(pair) in (tuple, list):
+            return tuple(map(self._extract_credential_value, pair))
```

Comment:
This returns a tuple of optional strings, which means your function signature is wrong.

I don't think a function that returns Optional[Tuple[Optional[str], Optional[str]] is great design. Can you improve?

Reply:
oof, yeah, that is not pretty. The type hint on the return is the intended output, I will clean that up.

Reply:
I guess the "can you add typehints" request from above is the reason it's not an error in mypy (?)

Reply:
all of this is removed, new check is `pair if all(pair) else None`